File: project/5.EX_総合理解度チェック/java/java_ex5_ex_2/src/java_ex5_ex_2/Ex5_ex_2.java
package java_ex5_ex_2;

class Trapezoid {
	double b1, b2, h;
	
	Trapezoid( double b1, double b2, double h ) {
		this.b1 = b1;
		this.b2 = b2;
		this.h  = h;
	}
	
	double getArea() {
		return ( this.b1 + this.b2 ) * this.h * 0.5;
	}
}

public class Ex5_ex_2 {

	public static void main(String[] args) {
		Trapezoid[] tp = new Trapezoid[3];
		
		tp[0] = new Trapezoid(2, 3, 4);
		tp[1] = new Trapezoid(3, 2, 2);
		tp[2] = new Trapezoid(4.4, 5.5, 6.6);
		
		System.out.println( tp[0].getArea() );
		System.out.println( tp[1].getArea() );
		System.out.println( tp[2].getArea() );

	}

}

File: project/5.EX_総合理解度チェック/py/py_ex5_ex_2/src/ex5_ex_2.py
'''
Created on 2023/05/05

@author: Keroichi.T
'''

class Trapezoid( object ):
    def __init__(self,b1, b2, h):
        self.b1 = b1
        self.b2 = b2
        self.h = h
        
    def getArea(self):
        return ( self.b1 + self.b2 ) * self.h * 0.5

def main():
    tp = [ 
           Trapezoid( 2, 3, 4 ),
           Trapezoid( 3, 2, 2 ),
           Trapezoid( 4.4, 5.5, 6.6 ),
         ]
    
    print( tp[0].getArea() )
    print( tp[1].getArea() )
    print( tp[2].getArea() )

if __name__ == '__main__':
    main()